### ISSUE TITLE
OPL4 sound devices should appear as 1

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#1529]
The OPL4 is implemented as 2 different sound devices in openMSX (FM and WAVE part). The consequence is that each device has its own volume setting. This is undesired, as it makes it possible to change the levels independently, which destroys the balance in music created for both of these parts (which will become more common, probably).

So, the volume control or sounddevice altogether, should appear as 1, just as it actually is.

Solution idea: could we make a new OPL4 sound device that has the 2 other sound chips as parts, and exposes a single volume control, which is used to set the volume of the sub parts? Or something like that :)

**Comment by MBilderbeek** [comment]
@m9710797 we now have an YMF278B device. Can that refactoring make this change easier?

**Comment by m9710797** [comment]
No, it doesn't make (much) difference. The new `YMF278B` class only aggregates the two `SoundDevice` sub-classes (`YMF262` (FM-part) and `YMF278` (wave-part)). So from a sound device point of view nothing changed.

At a technical level, modeling these two parts as two separate devices makes a lot of sense (it's a lot easier). E.g. they even operate at two different sample rates(*1). Actually that's really fundamental, in the current openMSX structure that makes it impossible to implement them as a single sound device.

From a user point of view it's mixed:
* I agree with you that changing the balance between the FM and wave part is not possible in real hardware. OTOH in openMSX it can be used to (temporarily) silence one but not the other part (rather than mute all the channels of one part).
* E.g. in the waveform viewer I personally think it's convenient that both parts are separate.

So is this request *only* about the shared vs separate volume setting? Or is there something more? If that's the only thing, then it might be possible to add a hack to share this single setting.

--

(*1) I vaguely remember there is some indication that the FM-part of the OPL4 (YMF278B) chip is redesigned to also operate at 44.1kHz (same as the wave part) instead of 49.7kHz (as the original OPL3 chip). And because of that there are subtle differences in various timings. Though I don't remember where I read this and whether it's true. For sure it's not how we currently emulate the FM-part, and I have also no idea how to adapt the emulation for it. (It will be a lot of work, for very little benefit).

**Comment by MBilderbeek** [comment]
The request is indeed only about showing it as unified, as in reality, it is only one chip. So, for a user, it is very weird to have it split into 2 audio devices with their respective settings/debuggables/etc.
